mcp/sensors/star_camera_solutions.c

```diff
@@ -0,0 +1,261 @@
+        first_time = 0;
+        sc2_rawtime_Addr = channels_find_by_name("sc2_rawtime");
+        sc2_ra_Addr = channels_find_by_name("sc2_ra");
+        sc2_dec_Addr = channels_find_by_name("sc2_dec");
+        sc2_image_rms_Addr = channels_find_by_name("sc2_image_rms");
+        sc2_fr_Addr = channels_find_by_name("sc2_fr");
+        sc2_ps_Addr = channels_find_by_name("sc2_ps");
+        sc2_ir_Addr = channels_find_by_name("sc2_ir");
+        sc2_alt_Addr = channels_find_by_name("sc2_alt");
+        sc2_az_Addr = channels_find_by_name("sc2_az");
+        sc2_photo_time_Addr = channels_find_by_name("sc2_photo_time");
+        sc2_ra_j2000_Addr = channels_find_by_name("sc2_ra_j2000");
+        sc2_dec_j2000_Addr = channels_find_by_name("sc2_dec_j2000");
+    }
+    SET_SCALED_VALUE(sc2_rawtime_Addr, scm.rawtime);
+    SET_SCALED_VALUE(sc2_ra_Addr, scm.ra_observed);
+    SET_SCALED_VALUE(sc2_dec_Addr, scm.dec_observed);
+    SET_SCALED_VALUE(sc2_image_rms_Addr, scm.image_rms);
+    SET_SCALED_VALUE(sc2_fr_Addr, scm.fr);
+    SET_SCALED_VALUE(sc2_ps_Addr, scm.ps);
+    SET_SCALED_VALUE(sc2_ir_Addr, scm.ir);
+    SET_SCALED_VALUE(sc2_alt_Addr, scm.alt);
+    SET_SCALED_VALUE(sc2_az_Addr, scm.az);
+    SET_SCALED_VALUE(sc2_photo_time_Addr, scm.photo_time);
+    SET_SCALED_VALUE(sc2_ra_j2000_Addr, scm.ra_j2000);
+    SET_SCALED_VALUE(sc2_dec_j2000_Addr, scm.dec_j2000);
+}
+
+static int which_sc(struct socket_data socket_setup) {
+    if (strcmp(socket_setup.ipAddr, SC1_IP_ADDR) == 0) {
+        return 1;
+    } else if (strcmp(socket_setup.ipAddr, SC2_IP_ADDR) == 0) {
+        return 2;
+    } else {
+        return -1;
+    }
+}
+
+static void where_to_unpack(struct mcp_astrometry data, int which_sc) {
+    if (which_sc == 1) {
+        assign_solution_data_to_channel_sc1(data);
+        return;
+    } else if (which_sc == 2) {
+        assign_solution_data_to_channel_sc2(data);
+        return;
+    } else {
+        blast_err("Invalid star camera address provided to decider");
+    }
+}
+
+// points the listening thread at the right status boolean to watch
+static int check_sc_thread_bool(int which) {
+    if (which == 1) {
+        return CommandData.sc_bools.sc1_image_bool;
+    } else if (which == 2) {
+        return CommandData.sc_bools.sc2_image_bool;
+    } else {
+        blast_err("Invalid star camera to check thread bool %d", which);
+        return -1;
+    }
+}
+
+static int check_for_reset(int which) {
+    if (which == 1 && CommandData.sc_resets.reset_sc1_image) {
+        // reset the flag
+        CommandData.sc_resets.reset_sc1_image = 0;
+        return 1;
+    } else if (which == 2 && CommandData.sc_resets.reset_sc2_image) {
+        CommandData.sc_resets.reset_sc2_image = 0;
+        return 1;
+    } else {
+        return 0;
+    }
+}
+
+void *image_receive_thread(void *args) {
+    struct socket_data * socket_target = args;
+    int first_time = 1;
+    int sockfd;
+    struct addrinfo hints, *servinfo, *servinfoCheck;
```

Comment:
Put variable declarations, especially if they're of different type, on separate lines.